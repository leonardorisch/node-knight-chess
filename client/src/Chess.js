import React, { Component } from 'react';
import './App.css';

class Chess extends Component {
  constructor(props) {
    super(props);
    this.darkClass = 'dark';
    this.lightClass = 'light';
  }

  render() {
    return (
      <div className="chessTable">
        {this.squareElements()}
      </div>
    );
  }

  squareElements() {
    const squares = []
    for(let x = 1; x < 9; x++) {
      for(let y = 1; y < 9; y++) {
        squares.push(
          <div className={"chessSquare " + this.squareClass(x, y)}
               key={x.toString() + y.toString()}> </div>
        )
      }
    }
    return squares
  }

  squareClass(x, y) {
    const odd = x % 2;

    if (y % 2) {
      return odd ? this.lightClass : this.darkClass
    }

    return odd ? this.darkClass : this.lightClass
  }
}

export default Chess;
