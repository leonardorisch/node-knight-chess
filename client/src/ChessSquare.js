import React, { Component } from 'react';
import axios from 'axios';

class ChessSquare extends Component {
  constructor(props) {
    super(props);
    this.darkClass = 'dark';
    this.lightClass = 'light';
    console.log(this.props)
    this.x = this.props.x;
    this.y = this.props.y;
    this.position = String.fromCharCode(this.x + 64) + this.y;
    this.selectSquare = this.selectSquare.bind(this);
    this.state = { active: '' }
  }

  render() {
    return <div className={`chessSquare ${this.squareClass(this.x, this.y)} ${this.state.active}`}
      onClick={this.selectSquare}>
    </div>
  }

  squareClass() {
    const odd = this.x % 2;

    if (this.y % 2) {
      return odd ? this.lightClass : this.darkClass;
    }

    return odd ? this.darkClass : this.lightClass;
  }

  selectSquare() {
    if(this.state.highlight) {
      this.setState({ highlight: '' });
    } else {
      const position = this.position;
      axios.post(`http://localhost:8626/knight`, {
        position: this.position
      })
      .then(res => {
        this.setState({ active: res.status === 200 ? 'highlight' : '' });
      })
    }
  }
}

export default ChessSquare;
