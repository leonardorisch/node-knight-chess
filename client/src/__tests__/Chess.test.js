import React from 'react';
import * as axios from "axios";
import Chess from '../Chess';
import ChessSquare from '../ChessSquare';
import "../setupTests";
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
jest.mock("axios");

test('renders without crashing', () => {
  const component = renderer.create(<Chess />)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('expect highlight component', () => {
  const table = mount(<Chess />);
  table.setState({ active: "A1" })
  expect(table.find('.highlight')).toHaveLength(1)
});

test('expect states state', () => {
  const table = shallow(<Chess />);
  axios.post.mockImplementation(() => Promise.resolve(
    jest.fn(table.setState({
      active: "A1", firstMovements: ['B3', 'C2'],
      secondMovements:["E1","E3","A1","A3","D4","B4","D2","D4","C1","C5","A1","A5"]
    }))
  ));
  const square = shallow(<ChessSquare position="A1" x="1" y="1"
  onActiveSquare={jest.fn()} />);

  square.simulate('click');
  expect(table.state().active).toEqual("A1");
  expect(table.state().firstMovements).toEqual(["B3", "C2"]);
  expect(table.state().secondMovements).toEqual(
    ["E1","E3","A1","A3","D4","B4","D2","D4","C1","C5","A1","A5"]
  );
});
