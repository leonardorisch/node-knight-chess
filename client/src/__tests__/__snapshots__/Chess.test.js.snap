// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="chessTable"
>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare dark"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="chessSquare light"
      onClick={[Function]}
    />
  </div>
</div>
`;
